audplot/core/api.py

```diff
@@ -818,11 +818,12 @@ def waveform(
             x_split.append(x[-rest:])
         else:
             x_split = np.split(x, splits)
-        samples = len(x_split)
         x_max = [x.max() for x in x_split]
         x_min = [x.min() for x in x_split]
+        x_axis = (np.arange(len(x_split))) * (samples / len(x_split))
+        samples = x_axis[-1] + 1
```

Comment:
Why do we need this line? Since we later set `xlim = (0, samples)` it means we possibly end up with a different size again.

Reply:
You are right, I removed that line.